### ISSUE TITLE
Allow skipping steps for ModelCheckpoint

### ISSUE THREAD

**Issue by rongcuid** [PyTorchLightning/pytorch-lightning#9498]
## 🚀 Feature

I would like `ModelCheckpoint` to skip the first `n` step for saving, either for k-best-model or for other uses.

### Motivation

My particular Seq2Seq model creates a validation loss graph like following:

![Screenshot from 2021-09-13 17-52-42](https://user-images.githubusercontent.com/1312525/133161469-3e3a254c-e649-411c-b548-e08b834b6db5.png)

Where you can see there is always an initial spike in the validation loss. This causes the top-k-model option in ModelCheckpoint to save k useless models coming from the beginning.

### Pitch

I would like an option like `skip_first_n=n` in `ModelCheckpoint`, such that I can, for example, skip attempting to checkpoint for the first 10k steps. Then, it should proceed as normal.

### Alternatives

I can subclass `ModelCheckpoint` to provide this functionality for my own use. However, I think this may be a common enough case that others may want it in Lightning.

### Additional context

**Comment by rongcuid** [comment]
Are there any style guides I should follow? This change by itself is just less than ten lines, so I will give it a try.

**Comment by rongcuid** [comment]
For instance, should these new params use `None`, `0`, or `-1` as default value meaning "no skipping"? Lightning seem to be quite inconsistent in this regard.

**Comment by rongcuid** [comment]
This is what I am doing now. I just think it might be a common thing.

**Comment by carmocca** [comment]
Closing this, we recommend https://github.com/PyTorchLightning/pytorch-lightning/issues/9498#issuecomment-936015637 going forward
